### ISSUE TITLE
Properly typed field name in NamedTuple causes Incompatible types in assignment error

### ISSUE THREAD

**Issue by rsiemens** [python/mypy#9043]
Hi there, :wave:, I ran into a strange issue when when type checking a `NamedTuple` with the specific field name, `count`. AFAICT it only affects this specific field name, but there may be others.

> Are you reporting a bug, or opening a feature request?

Bug

> Please insert below the code you are checking with mypy,

This simple mock up reproduces the issue

```python
from typing import NamedTuple

class Employee(NamedTuple):
    name: str
    count: int

employee = Employee(name="Test", count=5)
```

```
> python -m mypy foo.py
> foo.py:5: error: Incompatible types in assignment (expression has type "int", base class "tuple" defined the type as "Callable[[Tuple[object, ...], Any], int]")
Found 1 error in 1 file (checked 1 source file)
```

Using other name bindings works fine:

```python
from typing import NamedTuple

class Employee(NamedTuple):
    name: str
    age: int

employee = Employee(name="Test", age=5)
```

```
python -m mypy foo.py
Success: no issues found in 1 source file
```

Also using the `NamedTuple` constructor works fine:

```python
from typing import NamedTuple

Employee = NamedTuple("Employee", (("name", str), ("count", int)))

employee = Employee(name="Test", count=5)
```

```
python -m mypy foo.py
Success: no issues found in 1 source file
```

> What are the versions of mypy and Python you are using?

mypy 0.782
Python 3.8.1 

> Do you see the same issue after installing mypy from Git master?

Yes

> What are the mypy flags you are using? (For example --strict-optional)

default `python -m mypy foo.py`

**Comment by hauntsaninja** [comment]
Seems like the bug is caused by mypy getting confused by the methods NamedTuple inherits from tuple, so e.g., `index` fails as well.

**Comment by rsiemens** [comment]
Of course, that makes sense! I think then that this is actually not a bug since it's a legitimate thing to want to catch and warn about. Although it's a little strange the constructor version works fine.

I'm going to close this, but feel free to reopen if anyone feels otherwise. Thanks for your comments @hauntsaninja!

**Comment by funkyfuture** [comment]
see #4507 for an open issue in this regard.
